src/plugins/intel_gpu/src/graph/graph_optimizer/prepare_buffer_fusing.cpp

```diff
@@ -388,40 +455,17 @@ void prepare_buffer_fusing::run(program& p) {
                 if (p.is_loop_body() && node.get_dependency(0).is_type<lstm_elt>()) {
                     return;
                 }
-                // optimization is available for cropping across depth(features) only
+
+                // optimization is available for cropping across depth(features) or batch
                 // if output padding has defined padding across features already it wouldn't
                 // work because it expect to have zeros in the padded area.
-                const auto& crop_layout = node.get_output_layout();
-                auto format = crop_layout.format;
-                auto crop_prim = node.get_primitive();
-                auto input_layout = node.get_dependency(0).get_output_layout();
-                const auto& crop_size = crop_layout.get_tensor();
-                const auto& out_padd = crop_layout.data_padding;
-                auto opt_lower_pad = crop_prim->offsets.feature[0];
-                auto opt_upper_pad = input_layout.feature() - crop_prim->offsets.feature[0] - crop_size.feature[0];
-
-                // do not optimize crop if paddings are not properly aligned
-                for (auto& usr : node.get_users()) {
-                    auto usr_layout = usr->get_output_layout();
-                    if (usr_layout.format == format::b_fs_yx_fsv16 &&
-                        (opt_lower_pad % 16 != 0 || opt_upper_pad % 16 != 0))
-                        return;
-                    if (input_layout.data_padding.lower_size().batch[0] != 0 || input_layout.data_padding.upper_size().batch[0] != 0 ||
-                        input_layout.data_padding.lower_size().spatial[0] != 0 || input_layout.data_padding.upper_size().spatial[0] != 0 ||
-                        input_layout.data_padding.lower_size().spatial[1] != 0 || input_layout.data_padding.upper_size().spatial[1] != 0)
-                        return;
-                    // oneDNN doesn't support paddings
-                    if (usr->get_preferred_impl_type() == impl_types::onednn)
-                        return;
-                }
+                if (is_optimizable_padding_for_crop(node)) {
```

Comment:
what about style below:
```
if (!is_optimizable_padding_for_crop)())
   return;
```

Other code uses such style and it keeps indent small.

Reply:
Applied